Silicon/NXP/iMX6Pkg/Drivers/GOP/CPMem.c

```diff
@@ -0,0 +1,330 @@
+    cpmemWord0PackedReg.XVVirtualCoordinate = 0;
+    cpmemWord0PackedReg.YVVirtualCoordinate = 0;
+
+    // Subblock is unused although expect to have some value after write
+    cpmemWord0PackedReg.XBinnerBlockCoordinate = 0;
+    cpmemWord0PackedReg.YBinnerBlockCoordinate = 0;
+    cpmemWord0PackedReg.NewSubBlock = 0;
+
+    // Verify "current field" definition
+    cpmemWord0PackedReg.CurrentField = 0;
+
+    // Disable scrolling
+    cpmemWord0PackedReg.ScrollXCounter= 0;
+    cpmemWord0PackedReg.ScrollYCounter = 0;
+    cpmemWord0PackedReg.NumberOfScroll = 0;
+    cpmemWord0PackedReg.ScrollDeltaX = 0;
+    cpmemWord0PackedReg.ScrollMax = 0;
+    cpmemWord0PackedReg.ScrollingConfiguration = 0;
+    cpmemWord0PackedReg.ScrollingEnable = 0;
+    cpmemWord0PackedReg.ScrollDeltaY = 0;
+    cpmemWord0PackedReg.ScrollHorizontalDirection = 0;
+    cpmemWord0PackedReg.ScrollVerticalDirection = 0;
+
+    // Bits per pixel
+    cpmemWord0PackedReg.BitsPerPixel = FrameBufferPtr->Bpp;
+
+    // Decode Address Select
+    cpmemWord0PackedReg.DecodeAddressSelect = decodeAddressSelect;
+
+    // Scan order is progressive no support for interlace mode
+    cpmemWord0PackedReg.ScanOrder = CPMEM_SO_PROGRESSIVE;
+
+    // Band mode is sub frame double buffering
+    cpmemWord0PackedReg.BandMode = CPMEM_BNDM_DISABLE;
+
+    // Block mode used for post filtering and rotation
+    cpmemWord0PackedReg.BlockMode = CPMEM_BM_DISABLE;
+
+    // No support for rotation and flipping for now
+    cpmemWord0PackedReg.Rotation = CPMEM_ROT_NO_ROTATION;
+    cpmemWord0PackedReg.HorizontalFlip = CPMEM_HF_NO_HFLIP;
+    cpmemWord0PackedReg.VerticalFlip = CPMEM_HF_NO_VFLIP;
+    cpmemWord0PackedReg.ThresholdEnable = CPMEM_THE_DISABLE;
+
+    // Disable conditional access
+    cpmemWord0PackedReg.ConditionalAccessEnable = CPMEM_CAP_SKIP_LOW;
+    cpmemWord0PackedReg.ConditionalAccessPolarity = CPMEM_CAE_DISABLE;
+
+    // Frame width and height, minus one as 0 == 1 pixel
+    cpmemWord0PackedReg.FrameWidth = FrameBufferPtr->Width - 1;
+    cpmemWord0PackedReg.FrameHeight = FrameBufferPtr->Height - 1;
+
+    // No interrupt required at the end of line
+    cpmemWord0PackedReg.EndOfLineInterrupt = CPMEM_EOLI_DISABLE;
+
+    // Setting up CPMEM word 1
+    // Buffer 0, use the [31:3] bit address
+    cpmemWord1PackedReg.ExtMemBuffer0Address = FrameBufferPtr->PhyAddr >> 3;
+
+    // Buffer 1, use the same buffer for now
+    cpmemWord1PackedReg.ExtMemBuffer1Address = FrameBufferPtr->PhyAddr >> 3;
+
+    // Currently do not support interlace mode
+    cpmemWord1PackedReg.InterlaceOffset = 0;
+
+    // Pixel format and burst
+    cpmemWord1PackedReg.NumberOfPixelsInWholeBurstAccess = pixelBurst;
+    cpmemWord1PackedReg.PixelFormatSelect = pixelFormatSelector;
+
+    // Alpha config
+    cpmemWord1PackedReg.AlphaUsed = CPMEM_ALU_SAME_BUFFER;
+    // cpmemWord1PackedReg.AlphaChannelMapping = ???;
+
+    // TODO determine correct AXI id
+    cpmemWord1PackedReg.AxiId = CPMEM_ID_ID_0;

```

Comment:
AXI ID 0 should be okay, we don't have anything else contending for it.

Reply:
fixed comment